atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -184,12 +198,17 @@ boolean runOnce() {
                     progress.getFullTableName()),
                     batchSize,
                     progress.getStartRow());
+            long elapsedMillis = watch.elapsed(TimeUnit.MILLISECONDS);
             log.debug("Swept {} unique cells from {} starting at {}"
                     + " and performed {} deletions in {} ms"
                     + " up to timestamp {}.",
                     results.getCellsExamined(), progress.getFullTableName(),
                     progress.getStartRow() == null ? "0" : PtBytes.encodeHexString(progress.getStartRow()),
-                    results.getCellsDeleted(), watch.elapsed(TimeUnit.MILLISECONDS), results.getSweptTimestamp());
+                    results.getCellsDeleted(), elapsedMillis, results.getSweptTimestamp());
+            if (sweepPerfLogger.isPresent()) {
```

Comment:
I wonder if it would be easier to not pass an `Optional`, and instead implement a `DefaultBackgroundSweeperPerformanceLogger` that just prints the basics to debug, or possibly even a `NoOpBackgroundSweeperPerformanceLogger` if we really want to avoid any logging.